typescript/sdk/src/deploy/utils.ts

```diff
@@ -1,11 +1,25 @@
 import { ethers } from 'ethers';
 
+import { types } from '@abacus-network/utils';
+
 import { MultiProvider } from '../providers/MultiProvider';
-import { ChainName } from '../types';
+import { ChainMap, ChainName } from '../types';
 import { objMap } from '../utils';
 
 import { EnvironmentConfig } from './types';
 
+export function getMultiProviderFromConfigAndProvider<Chain extends ChainName>(
+  environmentConfig: EnvironmentConfig<Chain>,
+  provider: ethers.providers.Provider,
+): MultiProvider<Chain> {
+  const chainProviders = objMap(environmentConfig, (_, config) => ({
+    provider,
+    confirmations: config.confirmations,
+    overrides: config.overrides,
+  }));
+  return new MultiProvider(chainProviders);
+}
```

Comment:
iirc `EnvironmentConfig` has some multiprovider on the value type
this function seems only useful in testing when provider is consistent across chains and naming/typing should reflect this

Reply:
I don't follow sorry. This is useful for getting a multiprovider for checkers which don't need signers

Reply:
I believe @yorhodes 's point is that only in the testing environment, the provider is the same across every chain (which this function implicitly assumes)

Reply:
I see how it was easiest to just copy this from the function below. More generally, since this is used for testing only we can probably reduce it to something like `multiProviderFromTestProvider` and `multiProviderFromTestSigner` and then also actually assume `environmentConfig` to default to the configs for test1/test2/test3? That would probably simplify the interface quite well, though let me know if you find this to be out of scope, and I'm happy to file a ticket

Reply:
Will fix